File: Arrays/LongestConsecutiveSequence/longest_consecutive_sequence.dart
import 'dart:math';

void main(List<String> args) {
  print(longestConsecutiveSequence([2, 4, 3, 5, 3, 10, 12, 11, 16]));
}

longestConsecutiveSequence(List nums) {
  var set = Set();
  for (var i = 0; i < nums.length; i++) {
    set.add(nums[i]);
  }

  var maxLength = 0;

  for (var i = 0; i < nums.length; i++) {
    if (!set.contains(nums[i] - 1)) {
      var length = 0;
      while (set.contains(nums[i] + length)) {
        length++;
      }
      maxLength = max(maxLength, length);
    }
  }
  return maxLength;
}

File: Arrays/LongestConsecutiveSequence/longest_consecutive_sequence.js



function longestConsecutiveSequence(nums) {
    let set = new Set(nums);

    let max_length = 0;

    for (let i = 0; i < nums.length; i++) {
        if (!set.has(nums[i] - 1)) {
            length = 0;
            while (set.has(nums[i] + length)) {
                length++;
            }

            max_length = Math.max(max_length, length);
        }   
    }
    return max_length;
}

console.log(longestConsecutiveSequence([0,3,7,2,11,12,4,6,0,1]))